### PR TITLE
fix(webex-core): add default options interceptor

### PR DESCRIPTION
**Pull request by willward**
# Pull Request Template

## Description

This change adds a new Interceptor for setting default options for http-core request that will apply for every request made by the SDK. The consumer of the SDK can set the options when calling WebexSdk.init like this : 
```
webex = WebexSdk.init({
      credentials: {
        supertoken: superToken
      },
      config: {
        credentials: {
          client_id,
          client_secret
        },
        defaultRequestOptions: {
          timeout: 15000
        }
      }
    });
```
For the ediscovery downloader application we will use this feature to set : 
- a custom http agent for the request which performs ocsp revocation checking
- potentially a default request timeout

Fixes # (issue)
will be used to fix  https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-91060

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- add 3 new tests in `@webex/webex-core/test/unit/spec/interceptors/default-options.js`

**Test Configuration**:
* SDK Version - v1.79.0
* Node/Browser Version - v10.15.3
* NPM Version - v6.4.1

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@webex/webex-core/src/interceptors/default-options.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    const {defaultRequestOptions: defaultOptions} = this.webex.config;
+
+    if (!defaultOptions) {
+      return options;
+    }
+
+    Object.keys(defaultOptions).forEach((key) => {
+      // don't override any existing option properties
+      if (!Object.keys(options).includes(key)) {
+        options[key] = defaultOptions[key];
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/webex-core/src/interceptors/default-options.js]: Is there additional intent to append default header key:values in default options? If so, this may not work here, as it does not deep merge/check.

File: packages/node_modules/@webex/webex-core/src/interceptors/default-options.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+ *   webex = WebexSdk.init({
+ *     credentials: {
+ *       supertoken: superToken
+ *     },
+ *     config: {
+ *       credentials: {
+ *         client_id,
+ *         client_secret
+ *       },
+ *       defaultRequestOptions: {
```
Comment (InteractiveTimmy) [inline packages/node_modules/@webex/webex-core/src/interceptors/default-options.js]: I like how this can be implemented at the instance level. Might be able to shorten this to just `requestOptions`. Need additional input from the team.

File: packages/node_modules/@webex/webex-core/src/interceptors/default-options.js (hunk 1 above)
Comment (willward) [reply packages/node_modules/@webex/webex-core/src/interceptors/default-options.js]: the use cases i had was only for top level properties so didn't want to over complicate it, but if you think a deep merge would be better i can look into it.

File: packages/node_modules/@webex/webex-core/src/interceptors/default-options.js (hunk 2 above)
Comment (adamweeks) [reply packages/node_modules/@webex/webex-core/src/interceptors/default-options.js]: `defaultRequestOptions` works for me and is descriptive enough to not be confusing

**Comment by willward** [comment]
> All generated code looks good to me. My concern with an interceptor like this is that its scope may be too flexible. Additionally, other interceptors sometimes have conditionals that prevent them from modifying variables that already exist, so if the defaultOptions interceptor sets specific key:values, the requests will bypass the conditionals.

perhaps a whitelist of allowed properties that can be set by this interceptor would help to reduce the scope ?
